boards/posix/native_posix/doc/board.rst

```diff
@@ -185,4 +173,198 @@ with ``valgrind``.
+and some or all peripherals too.
+
+The simulator loads your binary, slowly interprets each instruction and
+accounts for the time each instruction takes.
+Time is simulated and is fully decoupled from real time.
+Simulations are in the order of x1/10 to x1/100 slower than real time.
+
+Some instruction set simulators will allow you to connect gdb to it, and may
+provide some extra tools for analyzing your code.
+
+Execution is fully reproducible. You can normally pause your execution without
+side-effects.
+
+
+Architecture
+************
+
+.. figure:: native_layers.svg
+    :align: center
+    :alt: Zephyr layering in native build
+    :figclass: align-center
+
+    Zephyr layering when built against an embedded target (left), and
+    targeting the native_posix board (right)
+
+This board is based on the POSIX architecture port of Zephyr.
+In this architecture each Zephyr thread is mapped to one POSIX pthread,
+but only one of these pthreads executes at a time.
+This architecture provides the same interface to the Kernel as other
+architectures and is therefore transparent for the application.
+
+This board does not try to emulate any particular embedded CPU or SOC.
+The code is compiled natively for the host x86 system, as a 32-bit
+binary assuming pointer and integer types are 32-bits wide.
+
+To ensure determinism when the Zephyr code is running,
+and to ease application debugging,
+the board uses a different time than real time: simulated time.
+This simulated time is, in principle, not linked to the host time.
+
+The Zephyr application sees the code executing as if the CPU were running at
+an infinitely high clock, and fully decoupled from the underlying host CPU
+speed.
+No simulated time passes while the application or kernel code execute.
+
+The CPU boot is emulated by creating the Zephyr initialization thread and
+letting it run. This in turn may spawn more Zephyr threads.
+Eventually the SW will run to completion, that is, it will set the CPU
+back to sleep.
+
+At this point, control is transferred back to the HW models and the simulation
+time can be advanced.
+
+When the HW models raise an interrupt, the CPU wakes back up, the interrupt
+is handled, the SW runs until completion again, and control is
+transferred back to the HW models, all in zero simulated time.
+
+If the SW unmasks a pending interrupt while running, or triggers a SW
+interrupt, the interrupt controller may raise the interrupt immediately
+depending on interrupt priorities, masking, and locking state.
+
+Normally the resulting executable runs fully decoupled from the real host time.
+That is, simulated time will advance as fast as it can. This is desirable when
+running in a debuger or testing in batch, but not if one wants to interact
+with external interfaces which are based on the real host time.
+
+Peripherals
+***********
+
+The following peripherals are currently provided with this board:
+
+**Console/printk driver**:
```

Comment:
This will display better if you write it as a definition list:

     **Console/printk driver**:
       A driver is provided that redirects any printk write to the native
       host application's stdout.

     **Simple timer**:
       A simple timer provides the kernel with a 10ms tick.
       This peripheral driver also provides the needed functionality for this
       architecture-specific :c:func:`k_busy_wait`.

       This timer, is ...

Reply:
Thank you very much David. All your corrections should be included now.